x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/search/action/SubmitAsyncSearchRequest.java

```diff
@@ -157,6 +157,15 @@ public boolean shouldCancelChildrenOnCancellation() {
                 // we cancel the underlying search action explicitly in the submit action
                 return false;
             }
+
+            @Override
+            public String getDescription() {
+                // generating description in a lazy way since source can be quite big
+                return "waitForCompletionTimeout[" + waitForCompletionTimeout +
+                    "], keepOnCompletion[" + keepOnCompletion +
+                    "] keepAlive[" + keepAlive +
+                    "], request=" + request.getDescription();
```

Comment:
`SubmitAsyncSearchRequest` should also override ` getDescription`  ?

Reply:
ehm, actually, what I did is not needed for submit, because we already provide the description to createTask, which is the result of submit#toString, which holds the same info that I meant to add. So nothing to do for submit unless we want to generate the description in a lazy way like search does.

Reply:
let's do it lazily since we print the request too ?